### PR TITLE
[FEATURE] Add a limit of events to process per run

### LINKED ISSUE #16: Add a limit of events to process per run
Default value to be defined (100 ?)

### PR DESCRIPTION
**Pull request by ghost**
Fix #16

### CODE REVIEW

File: Command/ProcessEventCommand.php [hunk 1]
```diff
@@ -40,6 +40,11 @@ class ProcessEventCommand extends ContainerAwareCommand
     private $channels;
 
     /**
+     * @var int
+     */
+    private $processedEventsCount = 0;
```
Comment (clement-michelet) [inline Command/ProcessEventCommand.php]: This variable should be local. The limit is per channel. The command can run multiple channel on the same run.

File: Command/ProcessEventCommand.php [hunk 2]
```diff
@@ -124,12 +129,13 @@ protected function execute(InputInterface $input, OutputInterface $output)
```
Comment (clement-michelet) [inline Command/ProcessEventCommand.php:124]: ~~You should initialize the local counter here.~~

File: DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -47,6 +47,8 @@ public function getConfigTreeBuilder()
                     ->useAttributeAsKey('name')
                     ->prototype('array')
                     ->children()
+                        ->scalarNode('events_limit_per_run')
+                            ->defaultValue(200)->end()
```
Comment (juchi) [inline DependencyInjection/Configuration.php]: I think the default behavior should be to process all events in queue, shouldn't it ?

**Comment by ghost** [comment]
ping @clement-michelet @juchi

**Comment by clement-michelet** [comment]
:+1:

File: Command/ProcessEventCommand.php [hunk 4]
```diff
@@ -122,14 +122,18 @@ protected function execute(InputInterface $input, OutputInterface $output)
 
             $this->lockHandler->lock($channel);
 
+            $processedEventsCount = 0;
             $event = null;
             try {
-                while (true) {
+                while (is_null($this->channels[$channel]['events_limit_per_run'])
```
Comment (juchi) [inline Command/ProcessEventCommand.php]: it's much faster to use `<expression> === null` rather than `is_null(<expression>)`

**Comment by juchi** [comment]
:+1: after the removal of is_null

**Comment by ghost** [comment]
@juchi: done!

**Pull request merged**